fcgi.go

```diff
@@ -0,0 +1,56 @@
+package exporter
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/mholt/caddy/caddyhttp/fastcgi"
+)
+
+// FastCGITransport is an implementation of RoundTripper that supports FastCGI.
+type FastCGITransport struct {
+	Network string
+	Address string
+}
+
+func (*FastCGITransport) timeout(req *http.Request) time.Duration {
+	t, ok := req.Context().Deadline()
+	if !ok {
+		return 0 // no timeout
+	}
+	return t.Sub(time.Now())
+}
+
+// RoundTrip implements the RoundTripper interface.
+func (t *FastCGITransport) RoundTrip(req *http.Request) (*http.Response, error) {
```

Comment:
Oh, I like the implementation as a roundtripper!

Reply:
Kudos to mackerel https://github.com/mackerelio/mackerel-agent-plugins/blob/master/mackerel-plugin-php-fpm/lib/fcgi.go